src/plots/plots.js

```diff
@@ -627,24 +627,22 @@ plots.createTransitionData = function(gd) {
 // whether a certain plot type is present on plot
 // or trace has a category
 plots._hasPlotType = function(category) {
-    // check plot
-    var basePlotModules = this._basePlotModules || [];
     var i;
 
+    // check base plot modules
+    var basePlotModules = this._basePlotModules || [];
     for(i = 0; i < basePlotModules.length; i++) {
-        var _module = basePlotModules[i];
-
-        if(_module.name === category) return true;
+        if(basePlotModules[i].name === category) return true;
     }
 
-    // check trace
+    // check trace modules
     var modules = this._modules || [];
-
     for(i = 0; i < modules.length; i++) {
-        var modulei = modules[i];
-        if(modulei.categories && modulei.categories.indexOf(category) >= 0) {
-            return true;
-        }
+        var name = modules[i].name;
+        if(name === category) return true;
+        // N.B. this is modules[i] along with 'categories' as a hash object
+        var _module = Registry.modules[name];
```

Comment:
TBH I've forgotten why we wrapped modules like this in the registry... not a big deal, but would it be better to attach the hash directly to the module, or even make `categories` as a hash from the beginning? Is it ever important that it's a list?

Reply:
Referencing https://github.com/plotly/plotly.js/issues/2174 where we like to make building custom trace modules more user friendly:

- we should try to **not** mutate the input module object, so that means not replacing the `categories` list with a hash object nor augmenting the input module object with e.g. a `categoryHash` key.
- listing categories as an array feels more natural than `{cat1: 1, cat2: 1}` hash object

So, I'm voting for the status quo.